in_toto/models/provenance.py

```diff
@@ -0,0 +1,220 @@
+#!/usr/bin/env python
+
+# Copyright New York University and the in-toto contributors
+# SPDX-License-Identifier: Apache-2.0
+
+"""
+<Program Name>
+  provenance.py
+
+<Author>
+  Furkan Türkal <furkan.turkal@trendyol.com>
+
+<Started>
+  Aug 3, 2021
+
+<Copyright>
+  See LICENSE for licensing information.
+
+<Purpose>
+  Provides a class for provenance metadata which is information gathered when a
+  step of the supply chain is performed.
+"""
+
+import attr
+import securesystemslib.formats
+from in_toto.models.common import Signable
+from in_toto.verifylib import _get_attribute
+
+FILENAME_FORMAT = "{step_name}.{keyid:.8}.provenance"
+FILENAME_FORMAT_SHORT = "{step_name}.provenance"
+UNFINISHED_FILENAME_FORMAT = ".{step_name}.{keyid:.8}.provenance-unfinished"
+UNFINISHED_FILENAME_FORMAT_GLOB = ".{step_name}.{pattern}.provenance-unfinished"
+
+
+@attr.s(repr=False, init=False)
+class Provenance(Signable):
```

Comment:
As we did in in-toto-golang, let's explicitly version predicate models. Perhaps just putting them in separate directories (and so packages) is sufficient.

Reply:
So ITE-6 predicates **require** us to also implement attestation Statements. I wonder if that belongs in this PR or if we add support for that separately and rebase. We should also have DSSE land soon which the Statement would build off? cc @lukpueh

Reply:
Also note that Statement support is mandatory because we need `subject`.